packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/utils.ts

```diff
@@ -51,24 +53,33 @@ const simpleBuyPaymentTypeDictionary = (rawPaymentType: SBPaymentTypes): Payment
 
 const getOriginalTimestamp = () => new Date().toISOString()
 
-const sha256 = (data) => crypto.createHash('sha256').update(data).digest()
-
 const generateUniqueUserId = (guid: string) => {
-  return sha256(guid).toString('base64')
+  return crypto.createHash('sha256').update(guid).digest().toString('base64')
 }
 
-const getNetworkFee = (value: PaymentValue | undefined) => {
-  return value
-    ? value.coin === 'BTC' || value.coin === 'BCH'
-      ? value.selection?.fee
-      : value.fee
+const getNetworkFee = (paymentValue: PaymentValue | null) => {
+  return paymentValue
+    ? paymentValue.coin === 'BTC' || paymentValue.coin === 'BCH'
+      ? paymentValue.selection?.fee
+      : paymentValue.fee
     : 0
 }
 
+const interestDepositClickedOriginDictionary = (rawOrigin) => {
+  switch (rawOrigin) {
+    case 'InterestPage':
+      return 'SAVINGS_PAGE'
+    default: {
+      return rawOrigin
+    }
```

Comment:
(minor) since there are only one case and default we might not need switch?

Reply:
As they are adding, I'm adding this one as a boilerplate